src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -327,14 +348,30 @@ JsonNode TransformSchemaNode(JsonSchemaExporterContext ctx, JsonNode schema)
 
                 if (!string.IsNullOrWhiteSpace(key.Description))
                 {
-                    ConvertSchemaToObject(ref schema).Insert(0, DescriptionPropertyName, (JsonNode)key.Description!);
+                    JsonObject obj = ConvertSchemaToObject(ref schema);
+                    JsonNode descriptionNode = (JsonNode)key.Description!;
+                    int index = obj.IndexOf(DescriptionPropertyName);
+                    if (index < 0)
+                    {
+                        obj.Insert(0, DescriptionPropertyName, (JsonNode)key.Description!);
+                    }
+                    else
+                    {
+                        obj[index] = (JsonNode)key.Description!;
+                    }
```

Comment:
There's no notion of just doing `obj[DescriptionPropertyName] = (JsonNode)key.Description!` and having it either update an existing property in place or add one?

Reply:
It's the latter. I wanted to add a comment about this but then forgot.